astropy/units/quantity_helper/erfa.py

```diff
@@ -98,6 +99,24 @@ def helper_gc2gd(f, nounit, unit1):
         )
 
 
+def helper_gc2gde(f, unit1, nounit, unit2):
+    from astropy.units.si import m, radian
+
+    if nounit is not None:
```

Comment:
I think that's too strict: it should just be `dimensionless_unscaled` (as a ratio of two radii). Note that this allows putting in float/ndarray - they will be treated as dimensionless. So this can just go inside the `try/except` below.

Reply:
Addressed in #14729, I wonder if the error message there will benefit from being more explicit though...